### ISSUE TITLE
unity 2018 crash issue 

### ISSUE THREAD

**Issue by FeeXue** [mixpanel/mixpanel-unity#63]
https://github.com/mixpanel/mixpanel-unity/blob/5d4ee9aff10b5fed62e91a16e85d6bb8c4f3452f/native/source/mixpanel/detail/persistence.cpp#L130

We are meeting a large numbers of crashes on our released app. And most of issues are pointing to Mixpanel. The the screenshot of Xcode displays the main issue about Mixpanel which is "mixpanel::detail::Persistence::persist_memory_queues()".

After initializing Mixpanel, we customized "queue size" and "flush_interval". The following code is the way to customize them:
Mixpanel.instance.set_maximum_queue_size(5 * 1024 * 1024);
Mixpanel.instance.set_flush_interval(10);

BTW, we are using the latest SDK "Aug 7th, 2018 - v1.3.1".

![20190701181828](https://user-images.githubusercontent.com/19380631/60559706-e59a1a80-9d7f-11e9-882a-cba47b55db86.jpg)

**Comment by FeeXue** [comment]
Is there any idea how to fix this issue?

**Comment by rocktavious** [comment]
@FeeXue I'm currently in the process of rewriting this library for mixpanel and in a month or two it should be ready for use on iOS (I would love to have you test it out when its ready to see if the rewrite has the same problems).  But my first guess would be to unset the max queue size override you have done to see if that fixes it.

**Comment by FeeXue** [comment]
> @FeeXue I'm currently in the process of rewriting this library for mixpanel and in a month or two it should be ready for use on iOS (I would love to have you test it out when its ready to see if the rewrite has the same problems). But my first guess would be to unset the max queue size override you have done (so it will use the default queue size) to see if that fixes it.

Cool, thanks you @rocktavious . From your first guess, you suggest me remove the to reset the queue size, but the max queue size is 5MB by default in mixpanel. But I only reset the flush time to 10 (60 by default). Does the flush time affect the memory in some way? 

Am I supposed to delete these two lines in my code:
> //Mixpanel.instance.set_maximum_queue_size(5 * 1024 * 1024);
> //Mixpanel.instance.set_flush_interval(10);
to keep "Max Queue Size" and "Flush_Interval" as default value?

**Comment by rocktavious** [comment]
@FeeXue The flush time shouldn't change anything - but i'm not 100% sure that it doesn't as the original library was written as a C native extension and I find it hard to follow/debug.  The release/2.0 branch (my rewrite in pure c#) will handle things correctly and should be compatible with you needs.  If you could try it out and let me know if you get the same problems. You can see the outstanding tasks in the 2.0 milestone here on github.  If you cannot i'd just comment out the `set_maximum_queue_size` and see if that helps.

**Comment by FeeXue** [comment]
I would like to try release/2.0 next time. 
For now, I plan to try to unset the "max queue size" and "flush interval" override like you said before.
And will let you know if the "unset" way works.

**Comment by rocktavious** [comment]
@FeeXue Did this work for you?

**Comment by FeeXue** [comment]
@rocktavious  Thanks for asking.:))))  Our QA is testing it, basically we did not find any crash related to mixpanel for now. But we are going to release the project in the middle of August. I will let you know what will happen in that project ASAP.

**Comment by FeeXue** [comment]
hi @rocktavious ,

Thank you for tracking this issue. 

If we just unset "max queue size" and "flush interval", we still have lots of crashes in our live app.
`
//Mixpanel.instance.set_maximum_queue_size(5 * 1024 * 1024);
//Mixpanel.instance.set_flush_interval(10);
`

And I plan to try release/2.0 to see if it works, and there are two question below:
- 1.Does release/2.0 fix the crash issue？
- 2.Does release/2.0 support Android 64bits?

By the way, one thing I am curious is, the date in CHANGELOG.md is April 12th - 2018, but the date in README.md is April 12th, 2019. These dates made me confused. 
![20190904130728](https://user-images.githubusercontent.com/19380631/64227344-b0dc4800-cf15-11e9-8424-d1065e01617b.jpg)
![20190904130748](https://user-images.githubusercontent.com/19380631/64227348-b6399280-cf15-11e9-8431-9fc23772941f.jpg)

Thanks,

**Comment by rocktavious** [comment]
@FeeXue Yes on both points - 2.0 shouldn't crash and also work on Android 64 bit.

Thanks for the date callout - i'll make sure its fixed once the release is full out.

**Comment by rocktavious** [comment]
2.0 has been released!
